### PR TITLE
Added phinx docs

### PR DESCRIPTION
**Pull request by lorenzo**
Copied documentation form sphinx repo here, so that we only have one place to update documentation and a single way for deploying it.

### CODE REVIEW

File: en/phinx/seeding.rst [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+        $ php vendor/bin/phinx seed:run -s UserSeeder
+
+You can also run multiple seeders:
+
+.. code-block:: bash
+
+        $ php vendor/bin/phinx seed:run -s UserSeeder -s PermissionSeeder -s LogSeeder
+
+You can also use the `-v` parameter for more output verbosity:
```
Comment (markstory) [inline en/phinx/seeding.rst]: These should be doubled up.

File: en/phinx/seeding.rst [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+----------------------
+
+This is the easy part. To seed your database, simply use the `seed:run` command:
+
+.. code-block:: bash
+
+        $ php vendor/bin/phinx seed:run
+
+By default, Phinx will execute all available seed classes. If you would like to
+run a specific class, simply pass in the name of it using the `-s` parameter:
```
Comment (markstory) [inline en/phinx/seeding.rst]: These should be doubled up.

File: en/phinx/seeding.rst [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+                      ->save();
+
+                // empty the table
+                $posts->truncate();
+            }
+        }
+
+.. note::
+
+    SQLite doesn't natively support the `TRUNCATE` command so behind the scenes
```
Comment (markstory) [inline en/phinx/seeding.rst]: These should be doubled up.

File: en/phinx/seeding.rst [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+Inserting Data
+--------------
+
+Using The Table Object
+~~~~~~~~~~~~~~~~~~~~~~
+
+Seed classes can also use the familiar `Table` object to insert data. You can
+retrieve an instance of the Table object by calling the ``table()`` method from
+within your seed class and then use the `insert()` method to insert data:
```
Comment (markstory) [inline en/phinx/seeding.rst]: The backticks in this paragraph need to be doubled up to enable inline-code.

File: en/phinx/configuration.rst [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+The first option specifies the path to your migration directory. Phinx uses
+``%%PHINX_CONFIG_DIR%%/db/migrations`` by default.
+
+.. note::
+
+    ``%%PHINX_CONFIG_DIR%%`` is a special token and is automatically replaced
+    with the root directory where your ``phinx.yml`` file is stored.
+
+In order to overwrite the default ``%%PHINX_CONFIG_DIR%%/db/migrations``, you
+need to add the following to the yaml configuration.
```
Comment (ravage84) [inline en/phinx/configuration.rst]: YAML in uppercase

File: en/phinx/configuration.rst [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+            charset: utf8
+            collation: utf8_unicode_ci
+
+would define a new environment called ``production``.
+
+In a situation when multiple developers work on the same project and each has
+a different environment (e.g. a convention such as ``<environment
+type>-<developer name>-<machine name>``), or when you need to have separate
+environments for separate purposes (branches, testing, etc) use environment
+variable `PHINX_ENVIRONMENT` to override the default environment in the yaml
```
Comment (ravage84) [inline en/phinx/configuration.rst]: YAML in uppercase

File: en/phinx/configuration.rst [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+
+Phinx currently supports the following database adapters natively:
+
+* `MySQL <http://www.mysql.com/>`_: specify the ``mysql`` adapter.
+* `PostgreSQL <http://www.postgresql.org/>`_: specify the ``pgsql`` adapter.
+* `SQLite <http://www.sqlite.org/>`_: specify the ``sqlite`` adapter.
+* `SQL Server <http://www.microsoft.com/sqlserver>`_: specify the ``sqlsrv`` adapter.
+
+SQLite
+`````````````````
```
Comment (ravage84) [inline en/phinx/configuration.rst]: Too long.

File: en/phinx/configuration.rst [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+    environments:
+        development:
+            adapter: sqlite
+            name: ./data/derby
+        testing:
+            adapter: sqlite
+            memory: true     # Setting memory to *any* value overrides name
+
+SQL Server
+`````````````````
```
Comment (ravage84) [inline en/phinx/configuration.rst]: Too long

File: en/phinx/configuration.rst [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+
+When using the ``sqlsrv`` adapter and connecting to a named instance you should
+omit the ``port`` setting as SQL Server will negotiate the port automatically.
+Additionally, omit the ``charset: utf8`` or change to ``charset: 65001`` which
+corresponds to UTF8 for SQL Server.
+
+Custom Adapters
+`````````````````
```
Comment (ravage84) [inline en/phinx/configuration.rst]: Too long

File: en/phinx/configuration.rst [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+The aliased classes will still be required to implement the ``Phinx\Migration\CreationInterface`` interface.
+
+.. code-block:: yaml
+
+    aliases:
+        permission: \Namespace\Migrations\PermissionMigrationTemplateGenerator
+        view: \Namespace\Migrations\ViewMigrationTemplateGenerator
+
+Version Order
+------
```
Comment (ravage84) [inline en/phinx/configuration.rst]: Too short

File: en/phinx/configuration.rst [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+
+When rolling back or printing the status of migrations, Phinx orders the executed migrations according to the 
+``version_order`` option, which can have the following values:
+
+* ``creation`` (the default): migrations are ordered by their creation time, which is also part of their filename.
+* ``execution``: migrations are ordered by their execution time, also known as start time.
```
Comment (ravage84) [inline en/phinx/configuration.rst]: Shouldn't there be meta stuff at the end of the file?

File: en/phinx/goals.rst [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+Goals
+=====
+
+Phinx was developed with the following goals in mind:
+
+* Be portable amongst the most popular database vendors.
+* Be PHP framework independent.
+* Have a simple install process.
+* Have an easy to use command-line operation.
+* Integrate with various other PHP tools (Phing, PHPUnit) and web frameworks.
```
Comment (ravage84) [inline en/phinx/goals.rst]: Shouldn't there be meta stuff at the end of the file?

**Comment by lorenzo** [comment]
@ravage84 thanks for the review, I didn't have time to dress all the issues you brought up, only the most important ones that I could action immediately.

I'll merge this to see how it looks like

**Pull request merged**